packages/cubejs-athena-driver/src/AthenaDriver.ts

```diff
@@ -124,16 +125,41 @@ export class AthenaDriver extends BaseDriver implements DriverInterface {
       config.secretAccessKey ||
       getEnv('athenaAwsSecret', { dataSource });
 
+    const assumeRoleArn = getEnv('athenaAwsAssumeRoleArn', { dataSource });
+    const assumeRoleExternalId = getEnv('athenaAwsAssumeRoleExternalId', { dataSource });
+
     const { schema, ...restConfig } = config;
 
     this.schema = schema ||
       getEnv('dbName', { dataSource }) ||
       getEnv('dbSchema', { dataSource });
 
+    // Configure credentials based on authentication method
+    let credentials;
+    if (assumeRoleArn) {
+      // Use assume role authentication
+      credentials = fromTemporaryCredentials({
+        params: {
+          RoleArn: assumeRoleArn,
+          ...(assumeRoleExternalId && { ExternalId: assumeRoleExternalId }),
+        },
+        // If access key and secret are provided, use them as master credentials
+        // Otherwise, let the SDK use the default credential chain (IRSA, instance profile, etc.)
+        ...(accessKeyId && secretAccessKey && {
+          masterCredentials: () => Promise.resolve({
```

Comment:
`masterCredentials` can be `AwsCredentialIdentity | AwsCredentialIdentityProvider`, let's use simple object here instead of doing async fn with this trick over Promise.resolve